crates/wasi-provider/src/states/initializing.rs

```diff
@@ -0,0 +1,138 @@
+use std::collections::HashMap;
+
+use log::{error, info};
+
+use crate::PodState;
+use k8s_openapi::api::core::v1::Pod as KubePod;
+use kube::api::{Api, PatchParams};
+use kubelet::container::{ContainerKey, Status as ContainerStatus};
+use kubelet::pod::{key_from_pod, Handle};
+use kubelet::state::prelude::*;
+
+use super::error::Error;
+use super::starting::{start_container, ContainerHandleMap, Starting};
+
+async fn patch_init_status(
+    client: &Api<KubePod>,
+    pod_name: &str,
+    name: String,
+    status: &ContainerStatus,
+) -> anyhow::Result<()> {
+    // We need to fetch the current status because there is no way to merge with a strategic merge patch ere
+    let mut init_container_statuses = match client.get(pod_name).await {
```

Comment:
So we actually tried this first. We have to have stable array elements to do so with a JSON Patch, which is possible here as we have to serially iterate through and run each container to completion. So it would technically be possible here, but not be possible in the `running` state because we'd be receiving updates from different containers within the pod at any time and would have to store an index mapping. That could be fine to do, but I didn't want to add yet another mapping to keep track of things right now